onnx/backend/test/case/node/resize.py

```diff
@@ -104,8 +132,170 @@ def export_downsample_linear():  # type: () -> None
         scales = np.array([1.0, 1.0, 0.6, 0.6], dtype=np.float32)
 
         output = np.array([[[
-            [1, 2.66666651]
+            [1.5, 3.5]
         ]]], dtype=np.float32)
 
         expect(node, inputs=[data, scales], outputs=[output],
                name='test_resize_downsample_linear')
+
+    @staticmethod
+    def export_resize_downsample_linear_align_corners():  # type: () -> None
+        node = onnx.helper.make_node(
+            'Resize',
+            inputs=['X', 'scales'],
+            outputs=['Y'],
+            mode='linear',
+            align_corners=1
+        )
+
+        data = np.array([[[
+            [1, 2, 3, 4],
+            [5, 6, 7, 8],
+        ]]], dtype=np.float32)
+
+        scales = np.array([1.0, 1.0, 0.6, 0.6], dtype=np.float32)
+
+        output = np.array([[[
+            [1, 4]
+        ]]], dtype=np.float32)
+
+        expect(node, inputs=[data, scales], outputs=[output],
+               name='test_resize_downsample_linear_align_corners')
+
+    @staticmethod
+    def export_resize_upsample_cubic():  # type: () -> None
+        node = onnx.helper.make_node(
+            'Resize',
+            inputs=['X', 'scales'],
+            outputs=['Y'],
+            mode='cubic',
+            align_corners=False
+        )
+
+        data = np.array([[[
+            [ 1,  2,  3,  4],
+            [ 5,  6,  7,  8],
+            [ 9, 10, 11, 12],
+            [13, 14, 15, 16],
+        ]]], dtype=np.float32)
+
+        scales = np.array([1.0, 1.0, 2.0, 2.0], dtype=np.float32)
+
+        output = np.array([[[
+            [ 0.47265625,  0.76953125,  1.24609375,  1.87500000,  2.28125000,
```

Comment:
Where do these numbers come from? It's not straightforward to write code to implement this behavior. Would you consider adding more `math` details and a lightweight numpy implementation here?

Reply:
@wschin 

These numbers come from pytorch and opencv. The result of opencv is the same as pytorch with align_corners=False.

I have implemented linear and cubic interpolation [here](https://gist.github.com/daquexian/18b3fda2c7d2a6ad236a0a5ded4fe845) :)

Reply:
Could you please have a Python version of your implementation (only depending on numpy) and compute the test outputs using that Python implementation? This operator is important and also very sophisticated, so a reference implementation is desired. Please take a look at #2066 for more reasons.

Reply:
All right, thanks for your suggestion :) I'll do it soon.